pkg/apis/monitoring/v1/types.go

```diff
@@ -1626,3 +1640,58 @@ func (l *PrometheusRuleList) DeepCopyObject() runtime.Object {
 type ProbeTLSConfig struct {
 	SafeTLSConfig `json:",inline"`
 }
+
+// AuthTLSConfig specifies TLS configuration parameters.
+// +k8s:openapi-gen=true
+type AuthTLSConfig struct {
+	SafeTLSConfig `json:",inline"`
+}
+
+// AuthCredentials allows to specify authorization credentials as either
+// BearerToken, BasicAuth, OAuth or TLS certificate. Fields are mutually exclusive.
+type AuthCredentials struct {
+	// Secret to read or mount
+	BearerTokenSecret v1.SecretKeySelector `json:"bearerTokenSecret,omitempty"`
+	// BasicAuth credentials.
+	BasicAuth *BasicAuth `json:"basicAuth,omitempty"`
+	// TODO add the rest too and validate, see prom/common
+	// config/http_config.go
+	// OAuth2 credentials. Only valid in Prometheus versions 2.27.0 and newer.
+	// OAuth2 *OAuth2 `json:"oauth2,omitempty"`
+	// TLSConfig for this Authorization section
+	// TLSConfig *AuthTLSConfig `json:"authTLSConfig,omitempty"`
+}
+
+// Authorization contains optional `Authorization` header configuration.
+// This section is only understood by  versions of Prometheus >= 2.26.0.
+// +k8s:openapi-gen=true
+type Authorization struct {
+	// Set the authentication type. Defaults to Bearer
+	Type string `json:"type,omitempty"`
+	// Secret to read or mount
+	Credentials *AuthCredentials `json:"credentials,omitempty"`
```

Comment:
Looking at the `authorization` section of the prometheus config, the `credentials` field is supposed to be a string. Which means that the type of `Credentials` should be `*v1.SecretKeySelector`.

Reply:
The intention of having some structure here is so that users don't have to specify the correct header value in a secret. We should also be able to re-use existing auth types here, say `OAuth2`.

Reply:
After quick call with @simonpasquier I see my error now. I was overthinking this quite a bit. Will simplify this.

Reply:
Sounds good, just let me know when you'd like me to look at the PR again :+1: